test/volumes-automount.test.js

```diff
@@ -0,0 +1,626 @@
+var TEST_IMAGE_LX = '7b5981c4-1889-11e7-b4c5-3f3bdfc9b88b'; // ubuntu-16.04
+var TEST_IMAGE_SMARTOS =
+    'ede31770-e19c-11e5-bb6e-3b7de3cca9ce'; // minimal-multiarch-lts (15.4.1)
+var UFDS_ADMIN_UUID = CONFIG.ufds_admin_uuid;
+
+// XXX
+TEST_IMAGE_SMARTOS = 'd0ebf524-2034-11e7-8e3d-878122f9de41';
+
+function deleteKeypair(cb) {
+    child_process.exec([
+        'rm',
+        '-f',
+        KEY_FILENAME,
+        KEY_FILENAME + '.pub'
+    ].join(' '), function onKeyPairDeleted(err, stdout, stderr) {
+        cb(err);
+    });
+}
+
+if (CONFIG.experimental_cloudapi_nfs_shared_volumes !== true) {
+    console.log('experimental_cloudapi_nfs_shared_volumes setting not ' +
+        'enabled, skipping tests');
+    process.exitCode = 0;
+} else {
+    var CLIENTS;
+    var CLIENT;
+    var SERVER;
+    var SSH_PUBLIC_KEY;
+
+    var fooContents;
+    var lxIp;
+    var lxVmUuid;
+    var networkUuidFabric;
+    var networkUuidSsh;
+    var smartosIp;
+    var smartosVmUuid;
+    var testPackage;
+    var testVolumeName = common.createResourceName('test-volumes-automount');
+    var testVolume;
+    var testVolumeStorageVmUuid;
+
+    function getMissingImages(t, missing, callback) {
+        if (missing.length === 0) {
+            callback();
+            return;
+        }
+
+        vasync.forEachParallel({
+            func: function importOneImage(imgUuid, cb) {
+                CLIENT.imgapi.adminImportRemoteImageAndWait(imgUuid,
+                    IMGAPI_SOURCE, {}, cb);
+            }, inputs: missing
+        }, function (err, results) {
+            callback(err);
+        });
+    }
+
+    test('setup', function (t) {
+        common.setup({clientApiVersion: '~8.0'}, function (_, clients, server) {
+            CLIENTS = clients;
+            CLIENT = clients.user;
+            SERVER = server;
+
+            t.end();
+        });
+    });
+
+    /*
+     * We need to provision on a network we can reach from cloudapi, so we use
+     * the first non-admin NIC on this cloudapi instance. That way we know the
+     * VM will be on the same network.
+     */
+    test('find external network', function (t) {
+        child_process.execFile('/usr/sbin/mdata-get', ['sdc:nics'],
+            function onExecOutput(execErr, stdout, stderr) {
+                var idx;
+                var nics;
+
+                t.ifErr(execErr, 'should be able to load sdc:nics');
+                nics = JSON.parse(stdout);
+
+                for (idx = 0; idx < nics.length; idx++) {
+                    if (networkUuidSsh === undefined &&
+                        nics[idx].nic_tag !== 'admin') {
+
+                        networkUuidSsh = nics[idx].network_uuid;
+                    }
+                }
+
+                t.ok(networkUuidSsh, 'found external network: '
+                    + networkUuidSsh);
+                t.end();
+            });
+    });
+
+    // need a package we can use to provision our containers
```

Comment:
Should that comment be about networks instead of packages?

Reply:
Indeed it should!